qemu/qemu.go

```diff
@@ -439,6 +475,14 @@ func (fsdev FSDevice) QemuParams(config *Config) []string {
 		deviceParams = append(deviceParams, fmt.Sprintf(",devno=%s", fsdev.DevNo))
 	}
 
+	if fsdev.DisableModern {
+		deviceParams = append(deviceParams, ",disable-modern=true")
+	} else {
+		deviceParams = append(deviceParams, ",disable-legacy=on")
+	}
+
+	deviceParams = append(deviceParams, ",iommu_platform=on")
+
```

Comment:
I don't know that you want to enable this unconditionally, since it can have a performance impact, and some virtio devices don't even properly support `iommu_platform=on` yet.

If you use `confidential-guest=support=` you won't need this anyway, since it automatically changes the qemu defaults so that iommu_platform will be enabled.